Test cases failures with UDB release 6.4

Some test cases fail with UDB release 6.4.

- [x] `TestNextParked`

    ```
    === RUN   TestNextParked
    --- FAIL: TestNextParked (1.54s)
        support.go:246: enabling recording for github.com/undoio/delve/pkg/proc_test.TestNextParked
        proc_test.go:81: recording
        proc_test.go:83: replaying "/tmp/undo911890882"
        proc_test.go:1965: Next did not continue on the selected goroutine, expected 6 got 15
    ```

- [x] `Test1ClientServer_exit`, `Test1NextGeneral`, `Test1NextFunctionReturn`, `TestIssue411`, `TestScopePrefix`, `TestOnPrefix`, `TestNoVars`, `TestOnPrefixLocals`, `TestIssue387`, `TestReverseContinue`, `TestCheckpoints`, `TestNextWithCount`, `TestIssue1090`, `TestPrintContextParkedGoroutine`, `TestStepOutReturn`, `TestOptimizationCheck`, `TestTruncateStacktrace`

    These cases all report "Internal debugger error: runtime error: slice bounds out of range [:-1]". I'll give one example.

    ```
    === RUN   Test1ClientServer_exit
    live-record: Maximum event log size is 1G
    Hello, World!
    live-record: Saving to '/tmp/undo518583002'...
    live-record: Termination recording written to /tmp/undo518583002
    live-record: Detaching...
    6.4.5 (build 17) udbserver
    using inferred keyfile: /home/grees/core/oldrelease-6.4.5.17/key
    Listening on port 34079
    Connection established
    --- FAIL: Test1ClientServer_exit (1.84s)
        support.go:246: enabling recording for github.com/undoio/delve/service/test_test.Test1ClientServer_exit
        integration1_test.go:177: Unexpected error: Internal debugger error: runtime error: slice bounds out of range [:-1]
            runtime.gopanic (0x430ca1)
                    /home/grees/goroot/src/runtime/panic.go:679
            runtime.goPanicSliceAlen (0x42f492)
                    /home/grees/goroot/src/runtime/panic.go:85
            github.com/undoio/delve/pkg/proc/gdbserial.(*Process).When (0x8d3051)
                    /home/grees/github.com/delve/pkg/proc/gdbserial/gdbserver.go:932
            github.com/undoio/delve/service/debugger.(*Debugger).state (0x8f2dd1)
                    /home/grees/github.com/delve/service/debugger/debugger.go:410
            github.com/undoio/delve/service/debugger.(*Debugger).State (0x8f29fe)
                    /home/grees/github.com/delve/service/debugger/debugger.go:367
            github.com/undoio/delve/service/rpc1.(*RPCServer).State (0x9031d7)
                    /home/grees/github.com/delve/service/rpc1/server.go:48
            reflect.Value.call (0x48fc05)
                    /home/grees/goroot/src/reflect/value.go:460
            reflect.Value.Call (0x48f3c3)
                    /home/grees/goroot/src/reflect/value.go:321
            github.com/undoio/delve/service/rpccommon.(*ServerImpl).serveJSONCodec.func2 (0x910a8a)
                    /home/grees/github.com/delve/service/rpccommon/server.go:318
            github.com/undoio/delve/service/rpccommon.(*ServerImpl).serveJSONCodec (0x90edf5)
                    /home/grees/github.com/delve/service/rpccommon/server.go:320
            runtime.goexit (0x460dd0)
                    /home/grees/goroot/src/runtime/asm_amd64.s:1357
    ```

- [x] `Test1ClientServer_breakpointInMainThread` and `TestRestart_breakpointPreservation` provoke null pointer dereference in the debugger. One example:

    ```
    === RUN   Test1ClientServer_breakpointInMainThread
    live-record: Maximum event log size is 1G
    Hello, World!
    ...
    live-record: Saving to '/tmp/undo768005806'...
    live-record: Termination recording written to /tmp/undo768005806
    live-record: Detaching...
    6.4.5 (build 17) udbserver
    using inferred keyfile: /home/grees/core/oldrelease-6.4.5.17/key
    Listening on port 36305
    Connection established
    --- FAIL: Test1ClientServer_breakpointInMainThread (2.22s)
        support.go:246: enabling recording for github.com/undoio/delve/service/test_test.Test1ClientServer_breakpointInMainThread
    panic: runtime error: invalid memory address or nil pointer dereference [recovered]
            panic: runtime error: invalid memory address or nil pointer dereference
    [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x923224]

    goroutine 83 [running]:
    testing.tRunner.func1(0xc000f74000)
            /home/grees/goroot/src/testing/testing.go:874 +0x3a3
    panic(0x9d7ce0, 0xf350b0)
            /home/grees/goroot/src/runtime/panic.go:679 +0x1b2
    github.com/undoio/delve/service/test_test.Test1ClientServer_breakpointInMainThread.func1(0xc000376540)
            /home/grees/github.com/delve/service/test/integration1_test.go:331 +0x2a4
    github.com/undoio/delve/service/test_test.withTestClient1.func1(0xc000376540, 0xa89bff, 0x8, 0xc0005b6920, 0x16, 0xc00002c780, 0x32, 0xc000226180, 0x26)
            /home/grees/github.com/delve/service/test/integration1_test.go:25 +0x2f
    github.com/undoio/delve/service/test_test.withTestClient1Extended(0xa89bff, 0x8, 0xc000f74000, 0xc00065bf10)
            /home/grees/github.com/delve/service/test/integration1_test.go:56 +0x404
    github.com/undoio/delve/service/test_test.withTestClient1(0xa89bff, 0x8, 0xc000f74000, 0xc00006af50)
            /home/grees/github.com/delve/service/test/integration1_test.go:24 +0x67
    github.com/undoio/delve/service/test_test.Test1ClientServer_breakpointInMainThread(0xc000f74000)
            /home/grees/github.com/delve/service/test/integration1_test.go:320 +0x80
    testing.tRunner(0xc000f74000, 0xab6918)
            /home/grees/goroot/src/testing/testing.go:909 +0xc9
    created by testing.(*T).Run
            /home/grees/goroot/src/testing/testing.go:960 +0x350
    ```

- [x] `TestCheckpoints`

    ```
    === RUN   TestCheckpoints
    live-record: Maximum event log size is 1G
    Hello, World!
    live-record: Saving to '/tmp/undo601200091'...
    live-record: Termination recording written to /tmp/undo601200091
    live-record: Detaching...
    6.4.5 (build 17) udbserver
    using inferred keyfile: /home/grees/core/oldrelease-6.4.5.17/key
    Listening on port 38991
    Connection established
    **************************************************************
    Assertion failure (ignored): gdbserial=0x2e307e8 gdbserial->fd=4 gdbserial->state=0, not gdbserial_state_RECEIVE
    Location: src/apps/udbserver/gdbserial.cpp:1068:gdbserial_internal_receive_raw [13656:13656]
    **************************************************************
    --- FAIL: TestCheckpoints (1.57s)
        support.go:246: enabling recording for github.com/undoio/delve/pkg/terminal.TestCheckpoints
        command_test.go:581: "continue": "> main.main() /home/grees/github.com/delve/_fixtures/continuetestprog.go:16 (hits goroutine(1):1 total:1) (PC: 0x48d2df)\n69a8\n    11:\t\n    12:\tfunc sayhi() {\n    13:\t\tfmt.Println(\"Hello, World!\")\n    14:\t}\n    15:\t\n=>  16:\tfunc main() {\n    17:\t\tsleepytime()\n    18:\t\tsayhi()\n    19:\t}\n"
        command_test.go:581: "next": "> main.main() /home/grees/github.com/delve/_fixtures/continuetestprog.go:17 (PC: 0x48d2eb)\n69a8\n    12:\tfunc sayhi() {\n    13:\t\tfmt.Println(\"Hello, World!\")\n    14:\t}\n    15:\t\n    16:\tfunc main() {\n=>  17:\t\tsleepytime()\n    18:\t\tsayhi()\n    19:\t}\n"
        command_test.go:581: "next": "> main.main() /home/grees/github.com/delve/_fixtures/continuetestprog.go:18 (PC: 0x48d2f0)\n6b66\n    13:\t\tfmt.Println(\"Hello, World!\")\n    14:\t}\n    15:\t\n    16:\tfunc main() {\n    17:\t\tsleepytime()\n=>  18:\t\tsayhi()\n    19:\t}\n"
        command_test.go:105: output of "restart c1": ""
        command_test.go:106: Error executing <restart c1>: protocol error E47 during restart for packet $vUDB;goto_time;69a8
    ```

--- a/pkg/proc/gdbserial/gdbserver.go
+++ b/pkg/proc/gdbserial/gdbserver.go
@@ -928,8 +928,7 @@ func (p *Process) When() (string, error) {
 		if err != nil {
 			return "", err
 		}
-		index := strings.Index(extent, ";")
-		result = extent[:index]
+		result = extent
 	} else {
 		event, err := p.conn.qRRCmd("when")
 		if err != nil {

--- a/pkg/proc/gdbserial/gdbserver_conn.go
+++ b/pkg/proc/gdbserial/gdbserver_conn.go
@@ -996,11 +996,11 @@ func (conn *gdbConn) restart(pos string) error {
 			// TODO: is defaulting to zero if we can't get it correct?
 			minBbCount := "0"
 
-			extent, err := conn.undoCmd("get_event_log_extent")
+			extent, err := conn.undoCmd("get_log_extent")
 			if err != nil {
 				return err
 			}
-			index := strings.Index(extent, ".")
+			index := strings.Index(extent, ",")
 			if index > 0 {
 				minBbCount = extent[:index]
 			}

